frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/add-feature-flag-modal/add-feature-flag-modal.component.ts

```diff
@@ -63,11 +64,13 @@ export class AddFeatureFlagModalComponent {
     public dialog: MatDialog,
     private formBuilder: FormBuilder,
     private featureFlagsService: FeatureFlagsService,
+    private experimentService: ExperimentService,
     private formHelpersService: CommonFormHelpersService,
     public dialogRef: MatDialogRef<AddFeatureFlagModalComponent>
   ) {}
 
   ngOnInit(): void {
+    this.experimentService.fetchContextMetaData();
```

Comment:
yeah I created an issue for the same, https://github.com/orgs/CarnegieLearningWeb/projects/1/views/12?pane=issue&itemId=67041859. I know we can just do this and it will work for now but architecturally it sort of doesn't make sense for "contextmetadata" to still be tied to the experiment service. Plus this cal is a little redundant if we already have that data.

But we can talk about it in meeting, I think we can merge this since we need it and keep my ticket open in case we feel like refactoring a little later.